### PR TITLE
[Explore] applying refresh chart overlay when chart is stale

### PR DESCRIPTION
**Pull request by gabe-lyons**
This is a follow up to apache#4459 to make it more obvious when the chart should be refreshed.

after making a change affecting the query:
image

dismissing the overlay still keeps the run button highlighted:
image

but if you make a change causing errors no overlay appears:
image

test plan:
- make a change to the query params
- verify the overlay appears
- verify run query works
- verify the overlay can be dismissed

reviewers:
@graceguo-supercat @michellethomas @mistercrunch

### CODE REVIEW

**Review by graceguo-supercat — CHANGES_REQUESTED** [review]
To me `refresh` means your new query generates new data, chart is inconsistent with latest query.

But when your control is messed up, you can not make a new query, so there is no new query results. `Refresh` chart doesn't make sense, and not necessary since old chart is still valid.

File: superset/assets/javascripts/components/RefreshChartOverlay.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        className="refresh-chart-overlay"
+      >
+        <div>
+          <Button 
+            className="query" 
+            onClick={this.props.onQuery} 
+            disabled={!!this.props.errorMessage}
+            bsStyle={this.props.errorMessage ? 'danger' : 'primary'}
+          >
+            <i className="fa fa-bolt" /> Refresh Chart
```
Comment (mistercrunch) [inline superset/assets/javascripts/components/RefreshChartOverlay.jsx]: i18n: `t('Refresh Chart')`

File: superset/assets/stylesheets/superset.less [hunk 2]
```diff
@@ -424,3 +428,12 @@ g.annotation-container {
     content: "\f0dd";
     color: @brand-primary;
 }
+
+.refresh-chart-overlay {
```
Comment (mistercrunch) [inline superset/assets/stylesheets/superset.less]: We could use a more general name if we want to reuse that style for the loading spinner

**Comment by GabeLoins** [comment]
@graceguo-supercat addressed the comments that came up in the team meeting- PTAL again!

**Review by graceguo-supercat — APPROVED** [review]

**Review by graceguo-supercat — DISMISSED** [review]
Can you fix linting error?

**Review by graceguo-supercat — APPROVED** [review]

**Pull request merged**